### ISSUE TITLE
tried loads following this on ubuntu

### ISSUE THREAD

**Issue by cr0wm4n** [IndrekHaav/photoprism-debian#9]
Hi
I have tried loads following this guide on ubuntu server, even using mariadb instead of sqlite3

No matter how many times i have started from a clean install, unfortuanely i can not get it to work.

Everything seems to install fine, but whenever i try to access the site i just get an error 500 :(

would you know if there is anything I can do to get it working without docker on ubuntu ?

Regards
Mike

**Comment by IndrekHaav** [comment]
Hard to say without knowing any actual details. What version of Ubuntu? Did you get any errors during the installation? Did you check the system logs and the output of `systemctl status photoprism`?

**Comment by cr0wm4n** [comment]
its the latest 22.04 server version.

and no errors in syslog or in journalctl and the staus shows it running.
it is a very strange one, i even tried connecting using lynx locally and still refuses to connect.
i disabled the ufw firewall to ensure it wasnt being blocked by that.
i tried several times full new installs of ubuntu both with and without the lamp stack so i could try both mariadb and sqlite3.

nothing has seemed to work and no clear cut errors anywhere :(

**Comment by cr0wm4n** [comment]
Im trying again with no lamp stack just plain sqlite3 version :)
Here is the 1st log from the make all command, it does have a couple of deprecated warnings but that is all, there are no errors that I can see...

mike@photos:~/photoprism$ sudo make all
scripts/download-facenet.sh
Installing Facenet model for TensorFlow...
Downloading latest model from https://dl.photoprism.app/tensorflow/facenet.zip?20221121...
--2022-11-21 21:11:13--  https://dl.photoprism.app/tensorflow/facenet.zip?20221121
Resolving dl.photoprism.app (dl.photoprism.app)... 5.9.32.79
Connecting to dl.photoprism.app (dl.photoprism.app)|5.9.32.79|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 87449984 (83M) [application/zip]
Saving to: ‘/tmp/photoprism/facenet.zip’

/tmp/photoprism/facenet.zip               100%[====================================================================================>]  83.40M  4.92MB/s    in 18s

2022-11-21 21:11:31 (4.74 MB/s) - ‘/tmp/photoprism/facenet.zip’ saved [87449984/87449984]

0492eb1d67789108b7eefb274e26633504b059be  /tmp/photoprism/facenet.zip
Archive:  /tmp/photoprism/facenet.zip
   creating: assets/facenet/
  inflating: assets/facenet/saved_model.pb
Latest Facenet installed.
scripts/download-nasnet.sh
Installing NASNet Mobile model for TensorFlow...
Downloading latest model from https://dl.photoprism.app/tensorflow/nasnet.zip?20221121...
--2022-11-21 21:11:32--  https://dl.photoprism.app/tensorflow/nasnet.zip?20221121
Resolving dl.photoprism.app (dl.photoprism.app)... 5.9.32.79
Connecting to dl.photoprism.app (dl.photoprism.app)|5.9.32.79|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 20682520 (20M) [application/zip]
Saving to: ‘/tmp/photoprism/nasnet.zip’

/tmp/photoprism/nasnet.zip                100%[====================================================================================>]  19.72M  4.76MB/s    in 4.3s

2022-11-21 21:11:36 (4.64 MB/s) - ‘/tmp/photoprism/nasnet.zip’ saved [20682520/20682520]

f18b801354e95cade497b4f12e8d2537d04c04f6  /tmp/photoprism/nasnet.zip
Archive:  /tmp/photoprism/nasnet.zip
   creating: assets/nasnet/
  inflating: assets/nasnet/labels.txt
   creating: assets/nasnet/variables/
  inflating: assets/nasnet/variables/variables.data-00000-of-00001
  inflating: assets/nasnet/variables/variables.index
  inflating: assets/nasnet/saved_model.pb
Latest NASNet Mobile installed.
scripts/download-nsfw.sh
Installing NSFW model for TensorFlow...
Downloading latest model from https://dl.photoprism.app/tensorflow/nsfw.zip?20221121...
--2022-11-21 21:11:36--  https://dl.photoprism.app/tensorflow/nsfw.zip?20221121
Resolving dl.photoprism.app (dl.photoprism.app)... 5.9.32.79
Connecting to dl.photoprism.app (dl.photoprism.app)|5.9.32.79|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 79188592 (76M) [application/zip]
Saving to: ‘/tmp/photoprism/nsfw.zip’

/tmp/photoprism/nsfw.zip                  100%[====================================================================================>]  75.52M  4.76MB/s    in 16s

2022-11-21 21:11:52 (4.73 MB/s) - ‘/tmp/photoprism/nsfw.zip’ saved [79188592/79188592]

2e03ad3c6aec27c270c650d0574ff2a6291d992b  /tmp/photoprism/nsfw.zip
Archive:  /tmp/photoprism/nsfw.zip
   creating: assets/nsfw/
 extracting: assets/nsfw/labels.txt
   creating: assets/nsfw/variables/
  inflating: assets/nsfw/variables/variables.data-00000-of-00001
  inflating: assets/nsfw/variables/variables.index
  inflating: assets/nsfw/saved_model.pb
Latest NSFW installed.
sudo npm install -g npm

removed 14 packages, changed 73 packages, and audited 223 packages in 2s

14 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
(cd frontend && npm ci --no-audit)
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated csurf@1.11.0: Please use another csrf package
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)

added 1174 packages in 14s

184 packages are looking for funding
  run `npm fund` for details
(cd frontend && env NODE_ENV=production npm run build)

> photoprism@1.0.0 build
> webpack --node-env=production

Starting PhotoPrism PRODUCTION build. Please wait.
10% building 0/1 entries 0/0 dependencies 0/0 modules(node:2678) [DEP_WEBPACK_COMPILATION_CACHE] DeprecationWarning: Compilation.cache was removed in favor of Compilation.getCache()
(Use `node --trace-deprecation ...` to show where the warning was created)
95% emitting emit OfflinePlugin(node:2678) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
        Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
        Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
assets by info 6.05 MiB [immutable]
  assets by chunk 761 KiB (auxiliary name: app)
    asset 698bd8550d863c20a3e1.ttf 339 KiB [emitted] [immutable] [from: src/css/vendor/icons/fonts/MaterialIcons-Regular.ttf] (auxiliary name: app)
    asset 08cefa1bc6566a650da7.woff 157 KiB [emitted] [immutable] [from: src/css/vendor/icons/fonts/MaterialIcons-Regular.woff] (auxiliary name: app)
    asset af9a28e7f261a412f581.eot 140 KiB [emitted] [immutable] [from: src/css/vendor/icons/fonts/MaterialIcons-Regular.eot] (auxiliary name: app)
    asset 2d8017489da689caedc1.woff2 122 KiB [emitted] [immutable] [from: src/css/vendor/icons/fonts/MaterialIcons-Regular.woff2] (auxiliary name: app)
    asset f64c3af3d0d25b9e4e00.svg 1.52 KiB [emitted] [immutable] [from: node_modules/photoswipe/dist/default-skin/default-skin.svg] (auxiliary name: app)
    asset 14d740b6ee64510747b0.gif 866 bytes [emitted] [immutable] [from: node_modules/photoswipe/dist/default-skin/preloader.gif] (auxiliary name: app)
    asset 0ff4d62150facee6b7de.png 547 bytes [emitted] [immutable] [from: node_modules/photoswipe/dist/default-skin/default-skin.png] (auxiliary name: app)
  assets by path *.js 4.88 MiB
    asset app.ba67dd004f8e873b61af.js 3.52 MiB [emitted] [immutable] [minimized] (name: app) 1 related asset
    asset share.78cc7a2e0495e5e783cd.js 1.36 MiB [emitted] [immutable] [minimized] (name: share)
  asset app.81f65a5a51e0137b6cf3.css 440 KiB [emitted] [immutable] (name: app)
asset sw.js 7.84 KiB [emitted]
asset assets.json 582 bytes [emitted]
Entrypoint app 3.95 MiB (761 KiB) = app.81f65a5a51e0137b6cf3.css 440 KiB app.ba67dd004f8e873b61af.js 3.52 MiB 7 auxiliary assets
Entrypoint share 1.36 MiB = share.78cc7a2e0495e5e783cd.js
orphan modules 2.34 MiB [orphan] 464 modules
runtime modules 2.23 KiB 11 modules
modules by path ./node_modules/ 3.64 MiB (javascript) 8.59 KiB (css/mini-extract)
  modules by path ./node_modules/core-js/ 524 KiB 502 modules
  modules by path ./node_modules/axios/ 56.8 KiB 32 modules
  modules by path ./node_modules/photoswipe/dist/ 113 KiB (javascript) 8.59 KiB (css/mini-extract)
    javascript modules 113 KiB 2 modules
    css modules 8.59 KiB 2 modules
  + 12 modules
modules by path ./src/ 3.47 MiB (javascript) 431 KiB (css/mini-extract)
  modules by path ./src/*.js 3.47 MiB
    ./src/app.js + 381 modules 2.2 MiB [built] [code generated]
    ./src/share.js + 51 modules 1.26 MiB [built] [code generated]
  ./src/common/navigation.js 310 bytes [built] [code generated]
  css ./node_modules/css-loader/dist/cjs.js??clonedRuleSet-2.use[1]!./node_modules/resolve-url-loader/index.js!./node_modules/postcss-loader/dist/cjs.js??clonedRuleSet-2.use[3]!./src/css/app.css 431 KiB [built] [code generated]
webpack 5.75.0 compiled successfully in 33240 ms
mike@photos:~/photoprism$

**Comment by cr0wm4n** [comment]
I had another warning at the end of the build.sh script 
so immediatly below #!/bin/bash  i added   
export CGO_CFLAGS="-g -O2 -Wno-return-local-addr"

there is a bug in go-sqlite3 as detailed here  
https://github.com/mattn/go-sqlite3/issues/803

I also enabled trace in the .env file, below is the output when i start the service.

Nov 21 21:52:40 photos systemd[1]: Starting PhotoPrism service...
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="config: defaults file /etc/photoprism/defaults.yml does not exist"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="config: running on 'Intel(R) Core(TM) i5-6600T CPU @ 2.70GHz', 8.2 GB memory detected"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="settings: loaded from /var/lib/photoprism/config/settings.yml"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="PhotoPrism® needs your support!"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="Visit https://photoprism.app/membership to learn more."
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="config: successfully initialized [2.300999ms]"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="migrate: executing pre migrations"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: found 2 previous migrations"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20221015-100000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20221015-100100 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="migrate: executing main migrations"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: found 11 previous migrations"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20211121-094727 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20211124-120008 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-040000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-050000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-061000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-071000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-081000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-083000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-091000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220329-093000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: 20220421-200000 skipped"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: errors migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: passwords migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_users migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: files_share migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: labels migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: services migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: duplicates migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: countries migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: reactions migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: migrations migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos_users migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: subjects migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: cells migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: cameras migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos_keywords migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_sessions migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: folders migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: files migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: files_sync migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: details migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: lenses migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: albums_users migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_users_settings migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos_albums migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: keywords migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_users_details migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: albums migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: links migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: places migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: categories migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos_labels migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: faces migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: markers migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_users_shares migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: completed in 80.77589ms"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="closed database connection"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="daemon started with process id 5843\n"
Nov 21 21:52:45 photos systemd[1]: Started PhotoPrism service.
mike@photos:/var/lib/photoprism$

and heres the output from systemctl status photoprism 

mike@photos:/var/lib/photoprism$ systemctl status photoprism.service
● photoprism.service - PhotoPrism service
     Loaded: loaded (/etc/systemd/system/photoprism.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2022-11-21 21:52:45 UTC; 6min ago
    Process: 5835 ExecStart=/opt/photoprism/bin/photoprism up -d (code=exited, status=0/SUCCESS)
   Main PID: 5843 (photoprism)
      Tasks: 9 (limit: 9271)
     Memory: 78.4M
        CPU: 10.517s
     CGroup: /system.slice/photoprism.service
             └─5843 /opt/photoprism/bin/photoprism up -d

Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: places migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: categories migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: photos_labels migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: faces migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: markers migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=trace msg="migrate: auth_users_shares migrated"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=debug msg="migrate: completed in 80.77589ms"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="closed database connection"
Nov 21 21:52:45 photos photoprism[5835]: time="2022-11-21T21:52:45Z" level=info msg="daemon started with process id 5843\n"
Nov 21 21:52:45 photos systemd[1]: Started PhotoPrism service.

So again, it looks like it starts fine

however, when i go to http://192.168.xxx.xxx:2342  i get forwarded to the /library/login page but again just get an error 500 :(

**Comment by cr0wm4n** [comment]
Got the bugga :)
it created a swapfile in the root called /swap.file
I changed the ownership of the swapfile to photoprism and now it starts :)
however it is incredibly slow and first time i logged in it did crash the service.
2nd time i logged in it is running but still feels very slow.
its an i5 6600t minimal ubuntu 22.04 server with 8gb ram that i am running it on

**Comment by cr0wm4n** [comment]
I have turned the logging back down to normal and not so slow now, but not as fast as i would like for a web app. I will play a bit more and see how it goes :)

**Comment by IndrekHaav** [comment]
Interesting. I haven't had this issue myself, though I've tested mainly in VMs, not directly on a physical machine.

Are you saying the /swap.file wasn't there before you installed Photoprism, but was there afterwards, and owned by root:root?

Is this a clean Ubuntu 22.04 install, or are you running something else on it as well? You mention the LAMP stack a couple of times, but that's not necessary for running Photoprism, it includes its own web server. If you want to reverse proxy it, then sure, Apache can do that, though something like Nginx might be a better choice.

What's the output of `free -h` ?

**Comment by cr0wm4n** [comment]
each install i tried is a fresh clean minimal install of ubuntu
The swap.file was there already, but photoprism could not access or make use of it.
I am looking for a better way to provide access to the swap file for photoprism currently.

as i said previous, i tried with and without the lamp stack, i currently do not have the lamp stack but will this time install just mariadb as it performs far better than sqlite3.

before that however i can see another issue, even with 8gb ram and a 4gb swapfile, after leaving it idling overnight i can now see i am suffering out of memory issues. 
I have nothing installed on the box except photoprism requirements and only uploaded 8 photographs to test with.

2 photoprisk processes got killed overnight due to out of memory.

right now the output of free -h is
               total        used        free      shared  buff/cache   available
Mem:           7.7Gi       6.8Gi       172Mi       1.0Mi       670Mi       594Mi
Swap:          4.0Gi       4.0Gi       1.0Mi

Regards
Mike

**Comment by IndrekHaav** [comment]
You shouldn't need to do anything to enable Photoprism to use the swap space, that's handled by the OS.

It definitely looks like something is using all available memory, forcing the system to use swap, which would cause the slowdown. Is it all being used by Photoprism?

My personal instance (running on Debian 11 in an LXC container) idles at around 500 MB, though I think I'm a few versions behind the latest. I've seen some posts on [r/photoprism](https://www.reddit.com/r/photoprism/) recently about high RAM usage and crashes, so maybe it's related to a recent update.

You could also try setting it up with Docker on the same machine, see if that makes a difference.

**Comment by cr0wm4n** [comment]
yes it is all being used by photoprism
there is almost nothing else on the box as it is a very clean new install
I have just added another 8GB of ram and set the ownership back correct on the swapfile
However, I am back to square 1, setting the ownership back correct has left me with the error 500 again.
It appears photoprism needs direct access to the swapfile as the photoprism user

as for docker, no thanks lol :)
im trying to get this working without docker as i do not want it or use it :)

**Comment by IndrekHaav** [comment]
I just set up a Debian 11 VM with 2 GB of RAM and a 1 GB swapfile:

```shell
$ ls -lh /swap*
-rw------- 1 root root 1.0G Nov 22 12:10 /swapfile
$ free -h
               total        used        free      shared  buff/cache   available
Mem:           1.9Gi       1.1Gi       282Mi       0.0Ki       529Mi       652Mi
Swap:          1.0Gi       4.0Mi       1.0Gi
```

Then I set up Photoprism and imported a few photos. Everything worked just fine, no crashes. I can see some swap usage, though not much; I'd probably need to import a lot more photos.

I have no idea why your system is behaving the way it is, sorry. I can try with Ubuntu 22.04 as well, but it'll take some time, I have to set up and install the VM from scratch (for Debian I can just clone a template on my Proxmox host).

Can you post a screenshot of the 500 error? Is there anything in the browser's Javascript console (F12 to open developer tools -> Console tab)? Anything in /var/log/syslog?

**Comment by cr0wm4n** [comment]
i have re-enabled the trace and changed the log so it logs to its own file
and can definately see its a swap file access issue

photoprism appears to start correctly as pointed out previously
however, every time i try to visit the site this appears in the trace log...
Notice the permission denied on the swapfile in the first line
if i change ownership of the swapfile to photoprism it then loads fine in the browser
its definately something to do with how its trying to use the swap file even now i have 16gb ram in the box

time="2022-11-22T12:54:15Z" level=debug msg="server: open /swap.img: permission denied (GET /library/login HTTP/1.1\r\nHost: 192.168.1.238:2342\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9\r\nAccept-Encoding: gzip, deflate\r\nAccept-Language: en-US,en;q=0.9\r\nCache-Control: max-age=0\r\nConnection: keep-alive\r\nDnt: 1\r\nUpgrade-Insecure-Requests: 1\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36\r\n\r\n)\n/usr/local/go/src/html/template/template.go:368 (0xa71c1b)\n\tMust: panic(err)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/render/html.go:71 (0xa71a39)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/render/html.go:61 (0xa74557)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/context.go:919 (0xab2457)\n/home/mike/photoprism/internal/server/routes_static.go:43 (0xe088f8)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/context.go:173 (0xe067f9)\n/home/mike/photoprism/internal/server/logger.go:20 (0xe067dc)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/context.go:173 (0xe06b0a)\n/home/mike/photoprism/internal/server/recovery.go:32 (0xe06af4)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/context.go:173 (0xab9ab0)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/gin.go:616 (0xab9718)\n/root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.1/gin.go:572 (0xab925c)\n/usr/local/go/src/net/http/server.go:2947 (0x7c63cb)\n\tserverHandler.ServeHTTP: handler.ServeHTTP(rw, req)\n/usr/local/go/src/net/http/server.go:1991 (0x7c15e6)\n\t(*conn).serve: serverHandler{

**Comment by cr0wm4n** [comment]
I have it working again :)
im not 100% sure what has fixed it
1st, i disabled the swap file and rebooted
2nd i then deleted the old swap file
3rd, i can see from the above log snippett that it is using 2 folders
       a:  /home/mike/photoprism/    (from where i installed it)
       b: /root/go/    ( i guess the build running the sudo commands placed this here)
I chown'ed both of those folders to photoprism
now it is starting fine without any swap enabled

by the way, do you have a email address, i have found something that may be of use to you

**Comment by cr0wm4n** [comment]
100% swap file issue, i recreated and enabled the swapfile again, and back to error 500 :)
i need to figure why its trying to use swap

**Comment by IndrekHaav** [comment]
Yeah, I haven't seen anything like that before. It's also strange that it's running from /root/go...

**Comment by cr0wm4n** [comment]
this is the script that installs go-tools in /root :)

https://raw.githubusercontent.com/photoprism/photoprism/develop/scripts/dist/install-go-tools.sh
Installing Go tools for X86_64 in /root/go...

I am going to look at something else now, i spent a lot of time these last few days looking at getting this installed without docker and there is several things i dont like for example. 
deprecated/unsupported packages are used
having to disable swap file
having it run from /root
and a few other things

thankyou for your efforts tho
Regards
Mike

**Comment by IndrekHaav** [comment]
Fair enough, good luck!
